cloudformation/aws-parallelcluster.cfn.json

```diff
@@ -2439,27 +2484,74 @@
                   "Sid": "S3BucketPolicy"
                 },
                 {
-                  "Sid": "DescribeInstances",
-                  "Effect": "Allow",
-                  "Action": [
-                    "ec2:DescribeInstances"
-                  ],
-                  "Resource": "*"
+                  "Fn::If": [
+                    "CreateHITSubstack",
+                    {
+                      "Sid": "DescribeInstances",
+                      "Effect": "Allow",
+                      "Action": [
+                        "ec2:DescribeInstances"
+                      ],
+                      "Resource": "*"
+                    },
+                    {
+                      "Ref": "AWS::NoValue"
+                    }
+                  ]
                 },
                 {
-                  "Sid": "FleetTerminatePolicy",
-                  "Effect": "Allow",
-                  "Action": [
-                    "ec2:TerminateInstances"
-                  ],
-                  "Resource": "*",
-                  "Condition": {
-                    "StringEquals": {
-                      "ec2:ResourceTag/Application": {
-                        "Ref": "AWS::StackName"
+                  "Fn::If": [
+                    "CreateHITSubstack",
+                    {
+                      "Sid": "FleetTerminatePolicy",
+                      "Effect": "Allow",
+                      "Action": [
+                        "ec2:TerminateInstances"
+                      ],
+                      "Resource": "*",
+                      "Condition": {
+                        "StringEquals": {
+                          "ec2:ResourceTag/Application": {
+                            "Ref": "AWS::StackName"
+                          }
+                        }
                       }
+                    },
+                    {
+                      "Ref": "AWS::NoValue"
                     }
-                  }
+                  ]
+                },
+                {
+                  "Fn::If": [
+                    "CreateHITSubstack",
+                    {
+                      "Sid": "Route53DeletePolicy",
+                      "Effect": "Allow",
+                      "Action": [
+                        "route53:ListResourceRecordSets",
+                        "route53:ChangeResourceRecordSets"
+                      ],
+                      "Resource": [
+                        {
+                          "Fn::Sub": [
+                            "arn:${AWS::Partition}:route53:::hostedzone/${ClusterHostedZone}",
```

Comment:
One problem here is that if we add a dependency on ComputeFleetHITSubstack, the lambda function performing the clean-up will only be created after ComputeFleetHITSubstack. This could be a problem in case the stack fails before ComputeFleetHITSubstack is created because the S3 bucket will be never removed. 
I'd say either we create a separate function or we get the name of the hosted zone from somewhere else that is available since the beginning